### PR TITLE
Make port of socks proxy configurable in product tests

### LINKED ISSUE #6182: Product tests fail if port 1080 is already used
When using Docker for Mac (not the older Docker Toolbox), product tests fail if port 1080 is used (which it always is for my SSH SOCKS proxy):

```
$ presto-product-tests/bin/run_on_docker.sh singlenode -g string_functions
...
Creating network "common_default" with the default driver
Creating common_hadoop-master_1

ERROR: for hadoop-master  Cannot start service hadoop-master: driver failed programming external connectivity on endpoint common_hadoop-master_1 (de2e876324560dff95b69e8cf4079aca9e9dab8d1e0f7db0039178e7a367b878): Error starting userland proxy: Bind for 0.0.0.0:1080 failed: port is already allocated
ERROR: Encountered errors while bringing up the project.
```

My random guess, knowing nothing about Docker, is that the old Docker setup used an isolated network for containers but the new way does bridging or uses the host network. 

If so, we should pick a different port for this, since using port 1080 is very common.

### PR DESCRIPTION
**Pull request by maciejgrzybek**
This fixes https://github.com/prestodb/presto/issues/6182.

### CODE REVIEW

File: presto-product-tests/conf/docker/common/compose-commons.sh [hunk 1]
```diff
@@ -40,3 +40,7 @@ if [[ -z ${PRODUCT_TESTS_JAR} ]]; then
     PRODUCT_TESTS_JAR="${PRODUCT_TESTS_ROOT}/target/presto-product-tests-${PRESTO_VERSION}-executable.jar"
 fi
 export PRODUCT_TESTS_JAR=$(canonical_path ${PRODUCT_TESTS_JAR})
+
+if [[ -z ${HIVE_PROXY_PORT} ]]; then
```
Comment (ArturGajowy) [inline presto-product-tests/conf/docker/common/compose-commons.sh]: Make it an oneliner:
`export HIVE_PROXY_PORT=${HIVE_PROXY_PORT:-1080}`

File: presto-product-tests/conf/docker/common/standard.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ services:
     image: ${HADOOP_MASTER_IMAGE}
     hostname: hadoop-master
     ports:
-      - '1080:1080'
+      - '${HIVE_PROXY_PORT}:1080'
```
Comment (ebd2) [inline presto-product-tests/conf/docker/common/standard.yml:15]: I believe you have this backwards. The first number is the internal port, the second is the one you're exposing it on.
This should read:

```
- '1080:${HIVE_PROXY_PORT}'
```

**Review by ebd2 — COMMENTED** [review]
LGTM mod comment

File: presto-product-tests/README.md [hunk 3]
```diff
@@ -407,3 +407,14 @@ been downloaded:
     # downloaded.
     presto-product-tests/conf/docker/<profile>/compose.sh pull
     ```
+
+## Possible issues
```
Comment (akshatnair) [inline presto-product-tests/README.md]: Make the heading "Debugging issue" unless there is already a debugging section, if yes then add to that.

File: presto-product-tests/conf/docker/common/standard.yml (hunk 2 above)
Comment (maciejgrzybek) [reply presto-product-tests/conf/docker/common/standard.yml:15]: I think it's right. I checked locally (to see what port is mapped) before pushing and it worked in that configuration. There's difference between docker-compose and docker syntax.

File: presto-product-tests/README.md (hunk 3 above)
Comment (maciejgrzybek) [reply presto-product-tests/README.md]: This is not limited to debugging only. You may not even be able to start tests if port 1080 is taken.

Comment (maciejgrzybek) [reply presto-product-tests/README.md]: Renamed to `Known issues` though.

**Comment by maciejgrzybek** [comment]
Closing in favor of https://github.com/prestodb/presto/pull/6271. Thanks!

**Pull request closed without merging**